From dc96d5eaf109bae8eaf4053f5c782cb0392e20ea Mon Sep 17 00:00:00 2001
From: Colm O hEigeartaigh <coheigea@apache.org>
Date: Tue, 22 Dec 2015 11:03:45 +0000
Subject: [PATCH] [SANTUARIO-436] - XmlWriterToTree calls DOM with invalid
 global namespace URI. Thanks to Clement Pellerin for the patch.

git-svn-id: https://svn.apache.org/repos/asf/santuario/xml-security-java/trunk@1721345 13f79535-47bb-0310-9956-ffa450edef68
---
 .../jcp/xml/dsig/internal/dom/XmlWriterToTree.java     | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/src/main/java/org/apache/jcp/xml/dsig/internal/dom/XmlWriterToTree.java b/src/main/java/org/apache/jcp/xml/dsig/internal/dom/XmlWriterToTree.java
index 8466babe90..e31e962f9b 100644
--- a/src/main/java/org/apache/jcp/xml/dsig/internal/dom/XmlWriterToTree.java
+++ b/src/main/java/org/apache/jcp/xml/dsig/internal/dom/XmlWriterToTree.java
@@ -87,6 +87,11 @@ public XmlWriterToTree(List<XmlWriter.ToMarshal<? extends XMLStructure>> marshal
 
     @Override
     public void writeStartElement(String prefix, String localName, String namespaceURI) {
+        if ("".equals(namespaceURI)) {
+            // Map global namespace from StAX to DOM
+            namespaceURI = null;
+        }
+
         Element newElem = factory.createElementNS(namespaceURI, DOMUtils.getQNameString(prefix, localName));
         if (nextSibling != null) {
             newElem = (Element)nextSibling.getParentNode().insertBefore(newElem, nextSibling);
@@ -143,6 +148,11 @@ public Attr writeAttribute(String prefix, String namespaceURI, String localName,
 
         Attr result = null;
         if (value != null) {
+            if ("".equals(namespaceURI)) {
+                // Map global namespace from StAX to DOM
+                namespaceURI = null;
+            }
+
             result = factory.createAttributeNS(namespaceURI, DOMUtils.getQNameString(prefix, localName));
             result.setTextContent(value);
             if (! (currentNode instanceof Element)) {
